### PR TITLE
Add port channel test plan

### PR DESCRIPTION
**Pull request by richardyu-ms**
The test will include two parts
1. Lag functionalities
   - Loadbalance
2. Lag SAI APIs
   - create/check/remove lag and lag member

### CODE REVIEW

File: doc/sai-ptf/lag_test_plan.md [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+   - Load balancing
+2. Lag SAI APIs
+   - create/check/remove lag and lag member
+
+
+# Basic SAI APIs and sample packets
+
+## APIs
+
+Create and lag member
```
Comment (StormLiangMS) [inline doc/sai-ptf/lag_test_plan.md:38]: create lag and lag member?

File: doc/sai-ptf/lag_test_plan.md [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+counter_results = sai_thrift_get_port_stats(client, port)
+
+counter_results["SAI_PORT_STAT_IF_IN_DISCARDS"],
+counter_results["SAI_PORT_STAT_IF_IN_DISCARDS"],
+counter_results["SAI_PORT_STAT_IF_IN_UCAST_PKTS"],
+counter_results["SAI_PORT_STAT_IF_OUT_UCAST_PKTS"]))
+```
+
+Add fdb entry
+**P.s.For fordwarding packet with vlan tag, fdb entry should enable the bv_id**
```
Comment (StormLiangMS) [inline doc/sai-ptf/lag_test_plan.md:63]: In SONiC case, T0 portchannel towards T1, there would be no Vlan.

File: doc/sai-ptf/lag_test_plan.md [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+```Python
+simple_udp_packet(eth_dst=dst_mac,
+                    eth_src=src_mac,
+                    dl_vlan_enable=True,
+                    vlan_vid=vlan_id,
+                    pktlen=104)
+```
+
+# Test suites
+## Test suite #1: PortChannel Loadbalanceing
```
Comment (StormLiangMS) [inline doc/sai-ptf/lag_test_plan.md:90]: For this one, maybe we can divide them to 2 testcases, one is focus on the load balance. The other is for add/delete operation, which has no pkt loss.

For the load balance, maybe we also need to test the algorithm selection for the hash.

File: doc/sai-ptf/lag_test_plan.md [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+sai_thrift_set_lag_member_attribute(
+                self.client,
+                self.lag1_member4,
+                ingress_disable=True,
+                egress_disable=True)
+```
+
+| Steps/Cases | Goal | Expect  |
+|-|-|-|
+| Create lag, add lag members, port1 - 4. Add FDB entry for lag, map with a MAC. | Create lag and member| lag, and member created|
```
Comment (StormLiangMS) [inline doc/sai-ptf/lag_test_plan.md:107]: The port channel which should be a L3 interface, we should add route to do the forwarding, not FDB I think. In our usage, L2 forwarding only among identical IP subnet in same Vlan and normally are downlinks to server.

File: doc/sai-ptf/lag_test_plan.md [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+| Send packet with.| Prepare to send from lag to VLAN.| Vlan and members have been created.|
+| Send packet on port0 to the lag by specifying lag mac as dest mac. 4 times .| Packet forwards on port equally.| Loadbalance on lag members.|
+| Every time, disable egress/ingress on one lag member, then send packet | Packet forwards on available ports equally.| Loadbalance on lag members.|
+| Every time, enable egress/ingress on one lag member, then send packet | Packet forwards on available ports equally.| Loadbalance on lag members.|
+| Every time, remove one lag member, then send packet | Packet forwards on available ports equally.| Loadbalance on lag members.|
+
+## Test suite #2: Ingress/Egreee disable
+Sample APIs
```
Comment (StormLiangMS) [inline doc/sai-ptf/lag_test_plan.md:115]: Could you elaborate more about usage of ingress and egress here?

File: doc/sai-ptf/lag_test_plan.md [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+```
+
+lag port list
+```Python
+sai_thrift_get_lag_attribute(
+                self.client, self.lag1, port_list=portlist)
+```
+
+| Steps/Cases | Goal | Expect  |
+|-|-|-|
```
Comment (StormLiangMS) [inline doc/sai-ptf/lag_test_plan.md:134]: Same.

File: doc/sai-ptf/lag_test_plan.md [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+| Add FDB entry for port4 map to a MAC. Create lag and add port4 as a member. | Create lag and member| lag, and member created|
+| Create VLAN and add VLAN member with port0 and port1.| Prepare to send from lag to VLAN.| Vlan and members have been created.|
+| Send packet on port1 with target mac on port4. | Forwarding from port1 to port4.| Receive packet on port4.|
+| Disable egress and ingress on lag member4. send packet | Packet dropped on port4| Packet drop.|
+| Enable lag egress and ingress. Send packet with VLAN tag on lag port4 with a new dest mac.|Packet flooding on VLAN members, port0 and port1.|Packet received.|
+
```
Comment (StormLiangMS) [inline doc/sai-ptf/lag_test_plan.md:140]: May consider the scale case, like how many PC we can have, and how many ports we can have in PC (I think in azure up to 4?), and is that necessary to have consecutive port numbers for one PC?

Comment (richardyu-ms) [reply doc/sai-ptf/lag_test_plan.md:140]: ok, the threshold and the maximum can be checked and should be controlled from the switch aspect (another sai module), those functions(not only PC, also including FDB, ACL, VLAN volume, etc ) will be tested in that component.

File: doc/sai-ptf/lag_test_plan.md (hunk 4 above)
Comment (richardyu-ms) [reply doc/sai-ptf/lag_test_plan.md:107]: Yes, this is a simple way to make a local port forward a packet to an egress port. I will try to check how to achieve this with SAI APIs.

**Pull request closed without merging**